### ISSUE TITLE
[7.x] Route caching doubles URI prefixes

### ISSUE THREAD

**Issue by rjorel** [laravel/framework#31712]
- Laravel Version: 7.0.0
- PHP Version: 7.4.3
- Database Driver & Version: MySQL 5.7

### Description:
When prefixes are used for routes, running `route:cache` doubles prefixes in URIs.

Example:
```php
Route::group(['prefix' => '/contacts'], function () {
    Route::get('/download', 'Contacts\DownloadController@download')->name('contacts.download');

    // ...
});
```
in `routes/web.php` file.
After running `route:cache`, I get with `route:list`:
```
|  | GET|HEAD | contacts/contacts/download | contacts.download | App\Http\Controllers\Contacts\DownloadController@download | web,auth |
```
but when I try to access page, I have an error:
![image](https://user-images.githubusercontent.com/12407789/75801687-d5599580-5d7b-11ea-9143-938b5d38ffec.png)

**Comment by driesvints** [comment]
Hmm, you're right. I can reproduce this. I'll look into this. Routing itself still works as expected though.

**Comment by AramZahedi** [comment]
I also have the same problem.
I wanted to post a new issue then I saw this one already posted.
Please consider fixing it in the next releases.

**Comment by driesvints** [comment]
We did a new release which fixes this.

**Comment by olegmelnik** [comment]
Because of this, model binding did not work correctly

**Comment by AramZahedi** [comment]
The problem persists when the nesting level gets more than 2.

**Steps to reproduce:**

1. Add this routes group to your web router file.

`Route::prefix('profile')->group(function () {
    Route::prefix('{user:username}')->group(function () {
        /*
         |===============================================
         |                  Block User
         |===============================================
         */
        Route::prefix('block')->middleware(['auth'])->group(function () {
            Route::post('/', 'HomeController@index')
                ->name('profile.block');
            Route::delete('/', 'HomeController@index')
                ->name('profile.unblock');
        });
    });
});`

2. Cache the routes in the command line.

3. Now get the routes list from the command line, and you will see this route:
`profile/{user:username}/block/profile/{user}/block`

It seems that this problem only appears when routes are nested. Please solve the problem, so it doesn't appear with any level of nesting.

**Comment by rjorel** [comment]
Thanks for the quick fix, it's working perfectly for me now. :blush:
